scripts/get-sampleapp-url.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+#########################################################################################################
+# Script to retrieve the websphere sample app url from the openshift routes in the sample app namespace
+#########################################################################################################
+
+import json
+import logging
+import os
+import subprocess
+import sys
+import time
+
+input = sys.stdin.read()
+input_json = json.loads(input)
+
+# json response format
+json_response = {"response": ""}
+
+# debugging file init
+DEBUGFILE = "/tmp/getsampleappurl.log"
+if os.path.exists(DEBUGFILE):
+    os.remove(DEBUGFILE)
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.FileHandler(DEBUGFILE)],
+)
+
+
+# method to retrieve json attribute - returns the default value if the key doesn't exist or exists but it is null
+def get_attribute(data, attribute, default_value):
+    return data.get(attribute) or default_value
+
+
+def returnFatalError(message, mustExit=False):
+    message = {"error": "Error: " + message, "sampleapp_url": sampleapp_url}
+    returnResponse(message, True)
+    if mustExit:
+        sys.exit(1)
+
+
+# return the string on the standard error
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+# formats the string returned as response
+# if isError is true the output of the string is returned on sterr instead of stout
+def returnResponse(responseObject, isError=False):
+    json_response_formatted = json.dumps(responseObject, indent=2)
+    logging.info("returnResponse - response JSON content:")
+    logging.info(json_response_formatted)
+    json_response["response"] = json.dumps(responseObject, indent=None)
+    json_response_string = json.dumps(json_response, indent=None)
+    if isError:
+        eprint(json_response_string)
+    else:
+        print(json_response_string)
+
+
+# default values
+SLEEP = 10  # seconds to sleep
+DEFAULTRETRIES = 30  # 30 x 5 = 30 secs = 5 mins
+
+# reset input variables
+KUBECONFIG = ""
+APPNAMESPACE = ""
+APPNAME = ""
+ATTEMPTS = 0
+
+input_json_formatted = json.dumps(input_json, indent=2)
```

Comment:
No need to dump here, you already loaded the json. To get the values use `input_json.get("kubeconfig", "")` you do not need your custom get_attribute function then.

Reply:
the dump of the json got in input is done here to log the JSON correctly formatted in the logfile, it isn't used in any other portion of the code

About the the load of the JSON values, the `get_attribute` custom function is used to set default values